WebTransportSendStream docs plus WebTransportReceiveStream and friends (#30486)
mdn/content: files/en-us/web/api/webtransport_api/index.md

"Reliable" means that transmission and order of data are guaranteed. That provides slower delivery (albeit faster than with WebSockets), and is needed in situations where reliability and ordering are important (such as chat applications, for example).

When using reliable transmission via streams you can also set the relative priority of different streams over the same transport.

### Unidirectional transmission

> To open a unidirectional stream from a user agent, you use the {{domxref("WebTransport.createUnidirectionalStream()")}} method to get a reference to a {{domxref("WebTransportSendStream")}}.
> The `sendorder` property in the object passed as an argument is a number that sets the relative priority of this stream compared to other streams using the transport (bytes are sent first on streams with a higher number).

Author: Revert - not supported yet for unidirectional, though is for bidirectional
```suggestion
To open a unidirectional stream from a user agent, you use the {{domxref("WebTransport.createUnidirectionalStream()")}} method to get a reference to a {{domxref("WritableStream")}}. From this you can {{domxref("WritableStream.getWriter", "get a writer")}} to allow data to be written to the stream and sent to the server.
```

```js
async function writeData() {

>   const stream = await transport.createUnidirectionalStream({
>     sendOrder: "999995555",
>   });

Author: Revert 
```suggestion
  const stream = await transport.createUnidirectionalStream();
```

}
```

> The returned {{domxref("WebTransportSendStream")}} object inherits from {{domxref("WritableStream")}}, and can be used to {{domxref("WritableStream.getWriter", "get a writer")}} that allows data to be written to the stream and sent to the server.
> It can also be used to get statistics and other information for the stream.
> 

Author: Revert

```suggestion
```

Note also the use of the {{domxref("WritableStreamDefaultWriter.close()")}} method to close the associated HTTP/3 connection once all data has been sent.

If the server opens a unidirectional stream to transmit data to the client, this can be accessed on the client via the {{domxref("WebTransport.incomingUnidirectionalStreams")}} property, which returns a {{domxref("ReadableStream")}} of {{domxref("WebTransportReceiveStream")}} objects. These can be used to read {{jsxref("Uint8Array")}} instances sent by the server.

> 
> > **Note:** In earlier versions of the specification `ReadableStream` and `WritableStream` were used as receive and send streams for unidirectional and bidirectional streams intead of `WebTransportReceiveStream` and `WebTransportSendStream`.

Author: Remove the note as this is only true for bidirectional streams. I'll leave the bit above because the [evidence is](https://searchfox.org/mozilla-central/source/dom/webtransport/api/WebTransport.h#148) that the incoming stream is a WebTransportReceiveStream

```suggestion
```
